File: Expo/constants/languages.ts
export const SUPPORTED_LANGUAGES = [
  'English',
  'Assamese',
  'Bodo',
  'Dogri',
  'Gujarati',
  'Hindi',
  'Kannada',
  'Kashmiri',
  'Konkani',
  'Maithili',
  'Malayalam',
  'Bengali',
  'Manipuri',
  'Marathi',
  'Nepali',
  'Odia',
  'Punjabi',
  'Sanskrit',
  'Santali',
  'Sindhi',
  'Tamil',
  'Telugu',
  'Urdu',
] as const;

export type SupportedLanguage = string;

export const LANGUAGE_LABELS: Record<string, string> = {
  English: 'अंग्रेज़ी',
  Assamese: 'অসমীয়া',
  Bodo: 'बड़ो',
  Dogri: 'डोगरी',
  Gujarati: 'ગુજરાતી',
  Hindi: 'हिन्दी',
  Kannada: 'ಕನ್ನಡ',
  Kashmiri: 'कश्मीरी',
  Konkani: 'कोंकणी',
  Maithili: 'मैथिली',
  Malayalam: 'മലയാളം',
  Bengali: 'বাংলা',
  Manipuri: 'মণিপুরী',
  Marathi: 'मराठी',
  Nepali: 'नेपाली',
  Odia: 'ଓଡ଼ିଆ',
  Punjabi: 'ਪੰਜਾਬੀ',
  Sanskrit: 'संस्कृत',
  Santali: 'संताली',
  Sindhi: 'सिंधी',
  Tamil: 'தமிழ்',
  Telugu: 'తెలుగు',
  Urdu: 'اردو',
};

export function getLanguageLabel(language: string): string {
  return LANGUAGE_LABELS[language] ?? language;
}

File: backend/app/languages.py
from __future__ import annotations

# Canonical language names mapped to IndicTrans language codes.
LANGUAGE_TO_CODE = {
    "English": "eng_Latn",
    "Assamese": "asm_Beng",
    "Bodo": "brx_Deva",
    "Dogri": "doi_Deva",
    "Gujarati": "guj_Gujr",
    "Hindi": "hin_Deva",
    "Kannada": "kan_Knda",
    "Kashmiri": "kas_Arab",
    "Konkani": "gom_Deva",
    "Maithili": "mai_Deva",
    "Malayalam": "mal_Mlym",
    "Bengali": "ben_Beng",
    "Manipuri": "mni_Mtei",
    "Marathi": "mar_Deva",
    "Nepali": "npi_Deva",
    "Odia": "ory_Orya",
    "Punjabi": "pan_Guru",
    "Sanskrit": "san_Deva",
    "Santali": "sat_Olck",
    "Sindhi": "snd_Arab",
    "Tamil": "tam_Taml",
    "Telugu": "tel_Telu",
    "Urdu": "urd_Arab",
}

# Dialect and alternate names that are normalized to canonical names above.
LANGUAGE_ALIASES = {
    # Hindi cluster dialects.
    "Awadhi": "Hindi",
    "Avadhi": "Hindi",
    "Bhojpuri": "Hindi",
    "Braj": "Hindi",
    "Bundeli": "Hindi",
    "Chhattisgarhi": "Hindi",
    "Garhwali": "Hindi",
    "Haryanvi": "Hindi",
    "Kumaoni": "Hindi",
    "Magahi": "Hindi",
    "Marwari": "Hindi",
    # Bengali cluster dialects.
    "Sylheti": "Bengali",
    "Chittagonian": "Bengali",
    # Urdu variants.
    "Dakhini": "Urdu",
    "Hyderabadi Urdu": "Urdu",
    # Kannada variants.
    "Tulu": "Kannada",
    "Kodava": "Kannada",
    # Common alternate spellings.
    "Bangla": "Bengali",
    "Oriya": "Odia",
    "Meitei": "Manipuri",
}
